extensions/blocks/amazon/edit.js

```diff
@@ -1,23 +1,261 @@
+							checked={ showPrice }
+							onChange={ () => setAttributes( { showPrice: ! showPrice } ) }
+						/>
+						<ToggleControl
+							label={ __( 'Show Purchase Button', 'jetpack' ) }
+							checked={ showPurchaseButton }
+							onChange={ () => setAttributes( { showPurchaseButton: ! showPurchaseButton } ) }
+						/>
+					</PanelBody>
+					<PanelColorSettings
+						title={ __( 'Color Settings', 'jetpack' ) }
+						colorSettings={ [
+							{
+								value: backgroundColor,
+								onChange: newBackgroundColor =>
+									setAttributes( { backgroundColor: newBackgroundColor } ),
+								label: __( 'Background Color', 'jetpack' ),
+							},
+							{
+								value: textColor,
+								onChange: newTextColor => setAttributes( { textColor: newTextColor } ),
+								label: __( 'Text Color', 'jetpack' ),
+							},
+							{
+								value: buttonAndLinkColor,
+								onChange: newButtonAndLinkColor =>
+									setAttributes( { buttonAndLinkColor: newButtonAndLinkColor } ),
+								label: __( 'Button & Link Color', 'jetpack' ),
+							},
+						] }
+					>
+						{
+							<ContrastChecker
+								{ ...{
+									isLargeText: false,
+									textColor: textColor,
+									backgroundColor: backgroundColor,
+								} }
+							/>
+						}
+					</PanelColorSettings>
+				</>
+			) }
+		</InspectorControls>
+	);
+
+	const blockPreview = () => {
+		const {
+			title,
+			authors,
+			artists,
+			actors,
+			manufacturer,
+			detailPageUrl,
+			listPrice,
+			imageUrlMedium,
+			imageWidthMedium,
+			imageHeightMedium,
+			authorshipInfo,
+		} = data.products.filter( productDataItem => productDataItem.asin === asin )[ 0 ];
+
+		// TODO - we should be able to get this from API in a neater way once we have access
+		const seller = authors.length
+			? authors
+			: artists.length
+			? artists
+			: actors.length
+			? actors
+			: manufacturer
+			? manufacturer
+			: authorshipInfo
+			? authorshipInfo
+			: null;
+
+		// TODO - we have different image sizes in the API
+		const image = imageUrlMedium && (
+			<a href={ detailPageUrl }>
+				<img
+					alt={ title }
+					src={ imageUrlMedium }
+					width={ imageWidthMedium }
+					heigth={ imageHeightMedium }
+				/>
+			</a>
+		);
+
+		const buttonTextColor = tinycolor
+			.mostReadable( buttonAndLinkColor, [ '#ffffff' ], {
+				includeFallbackColors: true,
+				size: 'small',
+			} )
+			.toHexString();
+
+		return (
+			asin && (
+				<div
+					style={ { backgroundColor: backgroundColor, color: textColor, width: imageWidthMedium } }
+				>
+					{ showImage && image }
+					{ showTitle && (
+						<div className={ `${ className }-title` }>
+							<a href={ detailPageUrl } style={ { color: buttonAndLinkColor } }>
+								{ title }
+							</a>
+						</div>
+					) }
+					{ showSeller && seller && (
+						<div className={ `${ className }-seller` }>
+							{ seller.length > 0 && typeof seller !== 'string'
+								? seller.map( singleSeller => singleSeller )
```

Comment:
Isn't this effectively a no-op? Should it be `seller.join( ', ' )` instead? 

Could we also massage the data before we get to this point? If we can get it ready for display and pass it into this preview component as props, it could make things cleaner.

Reply:
Yeah, as I say the whole concept of a seller will need to change when we are more sure what the API returns